### DISCUSSION TITLE
Infinite canvas code browsing - example.

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#348] [Ideas]
Just ran into this thread on HN: https://news.ycombinator.com/item?id=40279499

Link to product that has a quick visual demo:  https://territory.dev/

Interesting idea.  I am sure there are a lot of ideas like this around the various Lisp communities.

**Comment by tluyben** [comment]
I really like thinking about stuff like this and indeed it seems on more people have that (also in that HN thread). 

Small things like this help quick code understanding even for senior devs; 
![](https://racket-lang.org/img/ide-support.png) 
![](https://racket-lang.org/img/ugly-syntax.png)


For me it helps in another way; to keep functions small; I want to see this arrows on the same screen, so once it doesn't fit, they become less helpful which actually helps with better code practices imho. 

And indeed that canvas is quite nice, but (I tried it and similar tools) not *that* helpful in reality; I mean your IDE can also do this; it won't be on one screen but it didn't help me that much. 

There are a few opportunities for CLOG/CL that would indeed be very helpful; 

- when you context click any function, it can pop-over a bubble canvas like this Territory project with everything connected to that function in whatever file with previews -> VS code has that , but really not very nicely or helpfully implemented; you have to expand everything and everything is in a *very* small space, so you have no overview; you are still just looking a few lines at a time 

- something that would *really* be revolutionary but harder to do (but a longer term goal for me personally) is related to the previous thing; allow getting an overview like my previous point *during runtime*, so when i'm on a page in 'context mode' i can just ask about whatever widget to give me all related stuff; problem is, for that CLOG needs to be more widget based (which is what I am playing with anyway: flex/grid only layouts so it's easier to catch in robust, consistent LISP instead of html/css, widgets of that and then themes). But you can imagine the idea;  you ask for a button what it's context is and it uses it's powers to figure out the handlers, and so the code related to the handlers; you could see, on one screen/canvas, all logic, data, external calls etc related to that button 

Another nice thing  moldable tooling https://github.com/rabbibotton/clog/discussions/327 and such are great things to have which almost nothing actually has.

Strong IDE integration with a solid underlying language that allows introspection quite easily is needed for any of these things to actually be helpful.

**Reply by rabbibotton** [reply]
can you add your points as ticket(s) for feature enhancements

**Comment by khinsen** [comment]
That's a topic I am very much interested in for the future of my moldable inspector. It has a lot of code exploration facilities already (powered by `swank`), but there's always room for more. One idea I want to explore in the future is "tearing off" inspector panes and placing them on something like an infinite canvas.

Are there any examples available of an infinite canvas implemented in CLOG?
